src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -203,6 +227,24 @@ private async Task<T> GetConfigurationNonBlockingAsync(CancellationToken cancel)
 
                 try
                 {
+                    // Check if event handler can provide configuration
+                    if (ConfigurationEventHandler != null)
+                    {
+                        var configurationRetrieved = await HandleBeforeRetrieveAsync(cancel).ConfigureAwait(false);
```

Comment:
nit: can you update all the relevant comments in the event handler and property to clarify that if the handler fails to retrieve config, we will default to getting the config ourselves.